core/src/main/java/com/datastax/oss/driver/api/core/session/SessionBuilder.java

```diff
@@ -633,73 +637,103 @@ public SessionT build() {
+    DriverConfigLoader configLoader1 =
+        this.configLoader != null
+            ? this.configLoader
+            : defaultConfigLoader(programmaticArguments.getClassLoader());
+
+    DriverExecutionProfile defaultConfig = configLoader1.getInitialConfig().getDefaultProfile();
+    if (cloudConfigInputStream == null) {
+      String configUrlString =
+          defaultConfig.getString(DefaultDriverOption.CLOUD_SECURE_CONNECT_BUNDLE, null);
+      if (configUrlString != null) {
+        cloudConfigInputStream = () -> getURL(configUrlString).openStream();
       }
-      List<String> configContactPoints =
-          defaultConfig.getStringList(DefaultDriverOption.CONTACT_POINTS, Collections.emptyList());
-      if (cloudConfigInputStream != null) {
-        if (!programmaticContactPoints.isEmpty() || !configContactPoints.isEmpty()) {
-          throw new IllegalStateException(
-              "Can't use withCloudSecureConnectBundle and addContactPoint(s). They are mutually exclusive.");
-        }
-        String configuredSSLFactory =
-            defaultConfig.getString(DefaultDriverOption.SSL_ENGINE_FACTORY_CLASS, null);
-        if (sslConfigured || configuredSSLFactory != null) {
-          throw new IllegalStateException(
-              "Can't use withCloudSecureConnectBundle and explicitly specify ssl configuration. They are mutually exclusive.");
-        }
-        CloudConfig cloudConfig =
-            new CloudConfigFactory().createCloudConfig(cloudConfigInputStream.call());
+    }
+    List<String> configContactPoints =
+        defaultConfig.getStringList(DefaultDriverOption.CONTACT_POINTS, Collections.emptyList());
+    if (cloudConfigInputStream != null) {
+      if (!programmaticContactPoints.isEmpty() || !configContactPoints.isEmpty()) {
+        throw new IllegalStateException(
+            "Can't use withCloudSecureConnectBundle and addContactPoint(s). They are mutually exclusive.");
+      }
+      String configuredSSLFactory =
+          defaultConfig.getString(DefaultDriverOption.SSL_ENGINE_FACTORY_CLASS, null);
+      if (sslConfigured || configuredSSLFactory != null) {
+        throw new IllegalStateException(
+            "Can't use withCloudSecureConnectBundle and explicitly specify ssl configuration. They are mutually exclusive.");
+      }
+      try {
+        InputStream is = cloudConfigInputStream.call();
+        CloudConfig cloudConfig = new CloudConfigFactory().createCloudConfig(is);
         addContactEndPoints(cloudConfig.getEndPoints());
         withLocalDatacenter(cloudConfig.getLocalDatacenter());
         withSslEngineFactory(cloudConfig.getSslEngineFactory());
         withCloudProxyAddress(cloudConfig.getProxyAddress());
         if (cloudConfig.getAuthProvider().isPresent()) {
           withAuthProvider(cloudConfig.getAuthProvider().get());
         }
-        programmaticArguments = programmaticArgumentsBuilder.build();
+      } catch (IOException e) {
+        throw new UncheckedIOException(e);
```

Comment:
I'm changing this slightly so that this method does not throw checked exceptions; it seems overkill to impose to catch checked exceptions on callers of `buildLazy()`.